aixplain/processes/data_onboarding/process_audio_files.py

```diff
@@ -0,0 +1,201 @@
+__author__ = "thiagocastroferreira"
+
+import logging
+import os
+import pandas as pd
+import shutil
+import tarfile
+
+from aixplain.enums.file_type import FileType
+from aixplain.enums.storage_type import StorageType
+from aixplain.modules.file import File
+from aixplain.modules.metadata import MetaData
+from aixplain.utils.file_utils import upload_data
+from pathlib import Path
+from tqdm import tqdm
+from typing import List, Tuple
+
+
+def compress_folder(folder_path: str):
+    with tarfile.open(folder_path + ".tgz", "w:gz") as tar:
+        for name in os.listdir(folder_path):
+            tar.add(os.path.join(folder_path, name))
+    return folder_path + ".tgz"
+
+
+def run(metadata: MetaData, paths: List, folder: Path, batch_size: int = 100) -> Tuple[List[File], int, int, int]:
+    """Process a list of local audio files, compress and upload them to pre-signed URLs in S3
+
+    Explanation:
+        Each audio on "paths" is processed. If the audio is in a public link, this link is added into an index CSV file.
+        If the audio is in a local path, it will be copied into a local folder and its path will be added to the index CSV file.
+        The audios are processed in batches such that at each "batch_size" audios, the index CSV file is uploaded into a pre-signed URL in s3 and reset.
```

Comment:
Cut-paste an example that already exists for this kind of batched storage in the developer documentation.